runtime/elem/builtins/Core.h

```diff
@@ -221,6 +221,49 @@ namespace elem
         FloatType count = 0;
     };
 
+    // Counts the number of rising edges until a reset signal is received
+    template <typename FloatType>
+    struct Counter2Node : public GraphNode<FloatType> {
+        using GraphNode<FloatType>::GraphNode;
+
+        void process (BlockContext<FloatType> const& ctx) override {
+            auto** inputData = ctx.inputData;
+            auto* outputData = ctx.outputData[0];
+            auto numChannels = ctx.numInputChannels;
+            auto numSamples = ctx.numSamples;
+
+            // If we don't have the inputs we need, we bail here and zero the buffer
+            // hoping to prevent unexpected signals.
+            if (numChannels < 2)
+                return (void) std::fill_n(outputData, numSamples, FloatType(0));
+
+            for (size_t i = 0; i < numSamples; ++i) {
+                auto const in = inputData[0][i];
+                auto const reset = inputData[1][i];
+
+                // When reset is high, reset counter
+                if ((FloatType(1.0) - reset) <= std::numeric_limits<FloatType>::epsilon()) {
+                    count = FloatType(0);
+                }
+
+                // When the gate is high, count once
+                if ((FloatType(1.0) - in) <= std::numeric_limits<FloatType>::epsilon()) {
+                    if (!risingEdge) {
```

Comment:
If I'm following this right, any pulse that remains high for > 3 samples will record more than one count– the first 1.0 will count++ and set rising edge = true, the second 1.0 will set rising edge = false, the third 1.0 will set count++ and reset rising edge = true. So I think if this is meant to only count rising edges it will produce the wrong value.

There's a utility that you'll see used in this Core file called `Change` which reports the direction of signal change. With that you could do something like `if (floatEqual(FloatType(1.0), in) && change(in) > FloatType(0))` which would be true when the input signal is high _and_ the direction of change from last sample to this sample is also high (i.e. rising edge). That `change()` won't report another > 0 until the input signal goes low and then high again.